PrefectHQ/prefect-azure
# Add Azure Container Instances worker

This PR adds an Azure Container Instances worker.

### Steps to QA

Have an Azure account with an active subscription, or set one up. Then, on Azure:
* Create a service principal account
* Create a resource group

Then, using Prefect:
* Create an Azure Container Instance typed work pool
* Start an azure-container-instance worker
* Create a project that targets the work pool created above and sets (at a minimum) all the non-optional variables required by the worker
* Deploy the project
* Schedule or manually trigger flow runs and verify that they run to completion

### QA Completed
I have run many test flows on Azure using as many combinations of settings as I can. The worker seems to behave identically to the infrastructure block. 

### Docs
They exist:
<img width="922" alt="Screenshot 2023-04-20 050735" src="https://user-images.githubusercontent.com/228762/233320491-05e199ed-b934-484a-8b17-c9cfe7c9aede.png">

There's room to grow, but one of the key advantages of the worker over the ACI infra block (customizability via ARM template) is documented.

### Checklist



- [*] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-azure/issues/new/choose) first.
- [*] Includes tests or only affects documentation.
- [*] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [*] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [*] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-azure/blob/main/CHANGELOG.md)
